node/src/main/scala/org/bitcoins/node/networking/peer/ControlMessageHandler.scala

```diff
@@ -84,19 +80,39 @@ case class ControlMessageHandler(node: Node)(implicit ec: ExecutionContext)
     }
   }
 
-  def onPeerInitialization(peer: Peer): Future[Unit] = {
-    node.nodeAppConfig.nodeType match {
-      case nodeType: InternalImplementationNodeType =>
-        nodeType match {
-          case NodeType.FullNode =>
-            throw new Exception("Node cannot be FullNode")
-          case NodeType.NeutrinoNode =>
-            node.peerManager.createInDb(peer).map(_ => ())
+  def handleGossipAddrMessage(message: GossipAddrMessage): Unit = {
+    message match {
+      case addr: AddrMessage =>
+        addr.addresses.foreach { networkAddress =>
+          val bytes = Try(networkAddress.address.ipv4Bytes) match {
+            case Failure(_) =>
+              networkAddress.address.bytes
+            case Success(ipv4Bytes) =>
+              ipv4Bytes
+          }
+          val inetAddress =
+            NetworkUtil.parseInetSocketAddress(bytes, networkAddress.port)
+          val peer = Peer.fromSocket(socket = inetAddress,
+                                     socks5ProxyParams =
+                                       node.nodeAppConfig.socks5ProxyParams)
+          node.peerManager.addToFinder(peer)
         }
-      case nodeType: ExternalImplementationNodeType =>
-        nodeType match {
-          case NodeType.BitcoindBackend =>
-            throw new Exception("Node cannot be BitcoindBackend")
+      case addr: AddrV2Message =>
+        val bytes = addr.bytes
+        val port = addr.port.toInt
+        val inetAddress =
+          NetworkUtil.parseInetSocketAddress(bytes, port)
+        val peer = Peer.fromSocket(socket = inetAddress,
+                                   socks5ProxyParams =
+                                     node.nodeAppConfig.socks5ProxyParams)
+        addr match {
+          case IPv4AddrV2Message(_, _, _, _) | IPv6AddrV2Message(_, _, _, _) =>
+            //todo: ensure finder prioritises ones with filters
```

Comment:
Missed own todo. Shouldn't affect the remaining review. Right now even though can get peers that possibly support filters, the finder does not prioritizes trying them. Will fix this.